### ISSUE TITLE
What Are Smart Contracts? [Ultimate Beginner’s Guide to Smart Contracts]

### ISSUE THREAD

**Issue by neoscotch** [neoscotch/OpenRE#28]
**A Beginner’s Guide to Smart Contracts**

TLDR: A _smart contract_ is a computer protocol intended to digitally facilitate, verify, or enforce the negotiation or performance of a _contract_. _[Smart](https://blockgeeks.com/guides/different-smart-contract-platforms/) contracts_ allow the performance of credible transactions without third parties.

One of the best things about the [blockchain](https://blockgeeks.com/guides/what-is-blockchain-technology-a-step-by-step-guide-than-anyone-can-understand/) is that, because it is a decentralized system that exists between all permitted parties, there’s no need to pay intermediaries (Middlemen) and it saves you time and conflict. [Blockchains](https://blockgeeks.com/guides/video-guide-what-is-blockchain-technology/) have their problems, but they are rated, undeniably, faster, cheaper, and more secure than traditional systems, which is why banks and governments are turning to them.

## **Enjoy a free lesson from the Blockgeeks Library!**

![](https://embedwistia-a.akamaihd.net/deliveries/a4bdf195277dafe1184ce8e93870f1b7.webp?image_crop_resized=960x540)

In 1994, [Nick Szabo](https://en.wikipedia.org/wiki/Nick_Szabo), a legal scholar, and [cryptographer](https://blockgeeks.com/guides/cryptocurrencies-cryptography/) realized that the decentralized ledger could be used for smart contracts, otherwise called self-executing contracts, blockchain contracts, or digital contracts. In this format, contracts could be converted to computer code, stored and replicated on the system and supervised by the network of computers that run the blockchain. This would also result in ledger feedback such as transferring money and receiving the product or service.

## **What are Smart Contracts?**

Smart contracts help you exchange money, property, shares, or anything of value in a transparent, conflict-free way while avoiding the services of a middleman.

The best way to describe [smart contracts](https://blockgeeks.com/guides/decentralized-scalability/) is to compare the technology to a vending machine. Ordinarily, you would go to a lawyer or a notary, pay them, and wait while you get the document. With smart contracts, you simply drop a [bitcoin](https://blockgeeks.com/guides/what-is-bitcoin/) into the vending machine (i.e. ledger), and your escrow, driver’s license, or whatever drops into your account. More so, smart contracts not only define the rules and penalties around an agreement in the same way that a traditional contract does, but also automatically enforce those obligations. If you are looking for a more detailed walkthrough of smart contracts please check out our blockchain courses on smart contracts.

 ![](https://blockgeeks.com/wp-content/uploads/2019/05/smartcontractexplainer.jpg) 

![](https://blockgeeks.com/wp-content/uploads/2016/10/Vitalik-Buterin-150x150.png)

**As Vitalik Buterin, the 22-year-old programmer of [ethereum](https://blockgeeks.com/guides/ethereum/), explained it at a DC Blockchain Summit, in a smart contract approach, an asset or currency is transferred into a program “and the program runs this code and at some point it automatically validates a condition and it automatically determines whether the asset should go to one person or back to the other person, or whether it should be immediately refunded to the person who sent it or some combination thereof.”In the meantime, the decentralized ledger also stores and replicates the document which gives it a certain security and immutability.**

**_Example_**

Suppose you rent an apartment from me. You can do this through the blockchain by paying in [cryptocurrency](https://blockgeeks.com/guides/what-is-cryptocurrency-everything-you-need-to-know-ultimate-guide/). You get a receipt which is held in our virtual contract; I give you the digital entry key which comes to you by a specified date. If the key doesn’t come on time, the blockchain releases a refund. If I send the key before the rental date, the function holds it releasing both the fee and key to you and me respectively when the date arrives. The system works on the If-Then premise and is witnessed by hundreds of people, so you can expect a faultless delivery. If I give you the key, I’m sure to be paid. If you send a certain amount in bitcoins, you receive the key. The document is automatically canceled after the time, and the code cannot be interfered with either of us without the other knowing since all participants are simultaneously alerted.

You can use smart contracts for all sorts of situations that range from financial derivatives to insurance premiums, breach contracts, property law, credit enforcement, financial services, legal processes, and crowdfunding agreements.

## A Smart Contract Example

Here is the code for a basic smart contract that was written on the [Ethereum blockchain](https://blockgeeks.com/guides/what-is-ethereum/). Contracts can be encoded on any blockchain, but [ethereum](https://blockgeeks.com/guides/ethereum/) is mostly used since it gives unlimited processing capability.

[![](https://blockgeeks.com/wp-content/uploads/2016/10/Smart-contract-code-1.png)
](https://blockgeeks.com/wp-content/uploads/2016/10/Smart-contract-code-1.png)

_An example [smart contract on Ethereum](https://blockgeeks.com/ethereum-smart-contract-clients/). Source:_ _[https://www.ethereum.org/token](https://www.ethereum.org/token)_

The contract stipulates that the creator of the contract be given 10,000 BTCS (i.e. bitcoins); it allows anyone with enough balance to distribute these BTCs to others.

## **Here’s How You Can Use Smart Contracts**

![](https://blockgeeks.com/wp-content/uploads/2016/10/Jerry-Cuomo-1-1-150x150.png)

**Jerry Cuomo, vice president for [blockchain technologies](https://blockgeeks.com/guides/what-is-blockchain-technology/) at IBM, believes smart contracts can be used all across the chain from financial services to healthcare to insurance. Here are some examples:**

**_Government_**

Insiders vouch that it is extremely hard for our voting system to be rigged, but nonetheless, smart contracts would allay all concerns by providing an infinitely more secure system. Ledger-protected votes would need to be decoded and require excessive computing power to access. No one has that much computing power, so it would need God to hack the system! Secondly, smart contracts could hike low voter turnout. Much of the inertia comes from a fumbling system that includes lining up, showing your identity, and completing forms. With smart contracts, volunteers can transfer voting online and millennials will turn out en masse to vote for their Potus.

**_Management_**

The blockchain not only provides a single ledger as a source of trust, but also shaves possible snarls in communication and workflow because of its accuracy, transparency, and automated system. Ordinarily, business operations have to endure a back-and-forth, while waiting for approvals and for internal or external issues to sort themselves out. A blockchain ledger streamlines this. It also cuts out discrepancies that typically occur with independent processing and that may lead to costly lawsuits and settlement delays.

**Case history** 

In 2015, the Depository Trust & Clearing Corp. (DTCC) used a blockchain ledger to process more than $1.5 quadrillion worth of securities, representing 345 million transactions.

[**_Supply Chain_**](https://blockgeeks.com/guides/blockchain-and-supply-chain/)

Smart contracts work on the If-Then premise so, to put in [Jeff Garzik’s words](http://searchcio.techtarget.com/feature/What-is-a-smart-contract-and-whats-it-good-for),

![](https://blockgeeks.com/wp-content/uploads/2016/10/jeff-garzik.-Smart-Contracts-1-150x150.png)

**“UPS can execute contracts that say, ‘If I receive cash on delivery at this location in a developing, emerging market, then this other \[product], many, many links up the supply chain, will trigger a supplier creating a new item since the existing item was just delivered in that developing market.’” All too often, supply chains are hampered by paper-based systems, where forms have to pass through numerous channels for approval, which increases exposure to loss and fraud. The blockchain nullifies this by providing a secure, accessible digital version to all parties on the chain and automates tasks and payment.**

**Case history**

Barclays Corporate Bank uses smart contracts to log a change of ownership and automatically transfer payments to other financial institutions upon arrival

**_Automobile_**

There’s no doubt that we’re progressing from slothful pre-human vertebrates to super-smart robots. Think of a future where everything is automated. Google’s getting there with smartphones, smart glasses, and even smart cars. That’s where smart contracts help. One example is the self-autonomous or self-parking vehicles, where smart contracts could put into play a sort of ‘oracle’ that could detect who was at fault in a crash; the sensor or the driver, as well as countless other variables.  Using smart contracts, an automobile insurance company could charge rates differently based on where, and under which, conditions customers are operating their vehicles.

[**_Real Estate_**](https://blockgeeks.com/guides/blockchain-in-real-estate/)

You can get more money through smart contracts. Ordinarily, if you wanted to rent your apartment to someone, you’d need to pay a middleman such as Craigslist or a newspaper to advertise and then again you’d need to pay someone to confirm that the person paid rent and followed through. The ledger cuts your costs. All you do is pay via [bitcoin](https://blockgeeks.com/guides/what-is-bitcoin/) and encode your contract on the ledger. Everyone sees, and you accomplish automatic fulfillment. Brokers, [real estate](https://blockgeeks.com/guides/blockchain-real-estate/) agents, hard money lenders, and anyone associated with the property game can profit.

[**_Healthcare_**](https://blockgeeks.com/guides/blockchain-in-healthcare/)

Personal health records could be encoded and stored on the blockchain with a private key which would grant access only to specific individuals. The same strategy could be used to ensure that research is conducted via HIPAA laws (in a secure and confidential way). Receipts of surgeries could be stored on a blockchain and automatically sent to insurance providers as proof-of-delivery. The ledger, too, could be used for general healthcare management, such as supervising drugs, regulation compliance, testing results, and managing healthcare supplies.

 ![](https://blockgeeks.com/wp-content/uploads/2019/05/smartcontractexplainer.jpg) 

## Smart Contracts are Awesome!

**Here’s what smart contracts give you:**

**Autonomy –** You’re the one making the agreement; there’s no need to rely on a broker, lawyer or other intermediaries to confirm. Incidentally, this also knocks out the danger of manipulation by a third party, since execution is managed automatically by the network, rather than by one or more, possibly biased, individuals who may err.

**Trust** – Your documents are encrypted on a shared ledger.  There’s no way that someone can say they lost it.

**Backup** – Imagine if your bank lost your savings account. On the blockchain, each and every one of your friends have your back. Your documents are duplicated many times over.

**Safety** – [Cryptography](https://blockgeeks.com/guides/cryptocurrencies-cryptography/), the encryption of websites, keeps your documents safe. There is no hacking. In fact, it would take an abnormally smart hacker to crack the code and infiltrate.

**Speed** – You’d ordinarily have to spend chunks of time and paperwork to manually process documents. Smart contracts use software code to automate tasks, thereby shaving hours off a range of business processes.

**Savings** – Smart contracts save you money since they knock out the presence of an intermediary. You would, for instance, have to pay a notary to witness your transaction.

**Accuracy** – Automated contracts are not only faster and cheaper but also avoid the errors that come from manually filling out heaps of forms.

[![](https://blockgeeks.com/wp-content/uploads/2016/10/Smart-Contracts-are-Awesome-1.png)
](https://blockgeeks.com/wp-content/uploads/2016/10/Smart-Contracts-are-Awesome-1.png)

Here’s how Jeff Garzik, owner of blockchain services Bloq, described smart contracts:

> ## **“Smart contracts … guarantee a very, very specific set of outcomes.** **There’s never any confusion and there’s never any need for litigation.”**

* * *

**![](https://blockgeeks.com/wp-content/uploads/2016/10/Patrick-Hubbard-Head-Geek-SolarWinds-1-150x150.png)
“Smart Contracts are where the rubber meets the road for [businesses and blockchain](https://blockgeeks.com/guides/understand-blockchain-business-models/) technology. While a few highly specialized distributed financial services use cases for blockchain have appeared—for example, payment ledger services for the Yangon Stock Exchange in** **Myanmar. Its services on top of the blockchain that are really interesting. In the Yangon Exchange, it solves the problem of distributed settlement in a trading system that only synchronizes trades twice a day. But the autonomous execution capacities of smart contracts extend the transactional security assurance of blockchain into situations where complex, evolving context transitions are required. And it’s this possibility that has Amazon, Microsoft Azure and IBM Bluemix rolling out Blockchain-as-a-Service (Baas) from my butt.” –  Patrick Hubbard, Head Geek, SolarWinds**

## Now for Problems

Smart contracts are far from perfect. What if bugs get in the code? Or how should governments regulate such contracts? Or, how would governments tax these smart contract transactions?  As a case in point, remember my rental situation?

What happens if I send the wrong code, or, as lawyer [Bill Marino points out](http://tech.cornell.edu/news/smart-contracts-the-next-big-blockchain-application), I send the right code, but my apartment is condemned (i.e., taken for public use without my consent) before the rental date arrives? If this were the traditional contract, I could rescind it in court, but the blockchain is a different situation. The contract performs, no matter what.

The list of challenges goes on and on. Experts are trying to unravel them, but these critical issues do dissuade potential adopters from signing on.

## **And here’s To the Future of Smart Contracts…**

Part of the future of smart contracts lies in entangling these issues. In [Cornell Tech](http://tech.cornell.edu/news/smart-contracts-the-next-big-blockchain-application), for instance, lawyers, who insist that smart contracts will enter our everyday life, have dedicated themselves to researching these concerns.

Actually, when it comes to smart contracts, we’re stepping into a sci-fi screen. The IT resource center, Search Compliance suggests that smart contracts may impact changes in certain industries, such as law. In that case, lawyers will transfer from writing traditional contracts to producing standardized smart contract templates, similar to the standardized traditional contracts that you’ll find on LegalZoom. Other industries such as merchant acquirers, credit companies, and accountants may also employ smart contracts for tasks, such as real-time auditing and risk assessments. Actually, the website Blockchain Technologies sees smart contracts merging into a hybrid of paper and digital content where contracts are verified via blockchain and substantiated by physical copy.

## Blockchains Where You Can Process Smart Contracts

[Bitcoin](https://blockgeeks.com/bitcoin-transactions/): [bitcoin](https://blockgeeks.com/guides/what-is-bitcoin/) is great for processing Bitcoin transactions, but has limited ability for processing documents.

[Side Chains](http://gendal.me/2014/10/26/a-simple-explanation-of-bitcoin-sidechains/): This is another name for blockchains that run adjacent to [bitcoin](https://blockgeeks.com/guides/what-is-bitcoin/) and offer more scope for processing contracts.

NXT: NXT is a public blockchain platform that contains a limited selection of templates for smart contracts. You have to use what is given; you’re unable to code your own.

Ethereum: [ethereum](https://blockgeeks.com/guides/ethereum/) is a public blockchain platform and the most advanced for coding and processing smart contracts. You can code whatever you wish but would have to pay for computing power with “ETH” tokens.

As to the potential of smart contracts itself, there’s no end to the range of industries it can impact, from healthcare to automobiles to real estate and law. The list goes on and on. Says, [ethereum](https://blockgeeks.com/guides/ethereum/) CTO, Gavin Wood

![](https://blockgeeks.com/wp-content/uploads/2016/10/Gavin-Wood.-Smart-Contracts-1-150x150.png)

**“The potential for \[smart contracts] to alter aspects of society is of significant magnitude. This is something that would provide a technical basis for all sorts of social changes, and I find that exciting.**”

* * *

 [https://blockgeeks.com/guides/smart-contracts/](https://blockgeeks.com/guides/smart-contracts/)
